File: arrays-slices/605-can-place-flowers/can-place-flowers.go
package can_place_flowers

func canPlaceFlowers(flowerbed []int, n int) bool {
    if n == 0 {
      return true
    }
    var prev, next int
    for i, current := range flowerbed {
      prev, next = 0, 0
      if i>0 {
        prev = flowerbed[i-1]
      }
      if i < len(flowerbed) -1 {
        next = flowerbed[i+1]
      }
      if current == 0 && prev == 0 && next == 0 {
         n--
         if n==0 {
           return true
         }
         flowerbed[i] = 1 // помечаем как занятое место
      }
    }
    return false
}

File: arrays-slices/605-can-place-flowers/can-place-flowers.php
<?php


/**
 * @param Integer[] $flowerbed
 * @param Integer $n
 * @return Boolean
 * 1 0 0 0 0 0 1
 */
function canPlaceFlowers($flowerbed, $n)
{
    if ($n == 0) {
        return true;
    }
    $availableCount = 0;
    foreach ($flowerbed as $i => $v) {
        $prev = $flowerbed[$i - 1] ?? 0;
        $next = $flowerbed[$i + 1] ?? 0;
        if ($v == 0 && $prev == 0 && $next == 0) {
            $availableCount++;
            $flowerbed[$i] = 1; // помечаем клетку как занятую (т.к. мы ее уже посчитали)
            if ($availableCount >= $n) {
                return true;
            }
        }
    }
    return false;
}